### PR TITLE
Add specific project and Builds endpoints

### PR DESCRIPTION
**Pull request by nakhbari**
Keep the view models light for now, adding what we need as we need it.

#### /data/projects
`[{"project_name":"fastlane CI demo app tests","id":"db799377-aaa3-4605-ba43-c91a13c8f83"}]`

#### /data/projects/db799377-aaa3-4605-ba43-c91a13c8f83
`{"project_name":"fastlane CI demo app tests","id":"db799377-aaa3-4605-ba43-c91a13c8f83"}`

#### /data/projects/db799377-aaa3-4605-ba43-c91a13c8f83/builds
```
[{"number":18,"status":"failure"},{"number":17,"status":"failure"},{"number":16,"status":"failure"},{"number":15,"status":"failure"},{"number":14,"status":"failure"},{"number":13,"status":"pending"},{"number":12,"status":"pending"},{"number":11,"status":"pending"},{"number":10,"status":"pending"},{"number":9,"status":"pending"},{"number":8,"status":"pending"},{"number":7,"status":"pending"},{"number":6,"status":"pending"},{"number":5,"status":"pending"},{"number":4,"status":"pending"},{"number":3,"status":"pending"},{"number":2,"status":"pending"},{"number":1,"status":"pending"}]
```

#### /data/projects/db799377-aaa3-4605-ba43-c91a13c8f83/builds/18
`{"number":18,"status":"failure"}`

### CODE REVIEW

File: app/features-json/builds_json_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+module FastlaneCI
+  # Controller for providing all data relating to builds
+  class BuildsJSONController < AuthenticatedControllerBase
+    HOME = "/data/projects/:project_id/builds"
+
+    get HOME do |project_id|
+      # TODO: return NOT_FOUND if there is no project found
+      project = user_project_with_id(project_id: project_id)
+      builds_views_models = project.builds.map(&BuildSummaryViewModel.method(:viewmodel_from))
```
Comment (snatchev) [inline app/features-json/builds_json_controller.rb]: neat.

File: app/features-json/projects_json_controller.rb [hunk 2]
```diff
@@ -1,9 +1,10 @@
 require_relative "../shared/authenticated_controller_base"
 require_relative "./view_models/project_summary_view_model"
+require_relative "./view_models/project_view_model"
 
 module FastlaneCI
   # Controller for providing all data relating to projects
-  class ProjectJSONController < AuthenticatedControllerBase
+  class ProjectsJSONController < AuthenticatedControllerBase
```
Comment (KrauseFx) [inline app/features-json/projects_json_controller.rb]: Ah nice catch. This means we should also change all the other controllers, like `BuildController`, right?

Comment (taquitos) [reply app/features-json/projects_json_controller.rb]: @KrauseFx in this case, I think the controller manages multiple `project` objects, whereas the `BuildController` only deals with 1 at a time.

Comment (taquitos) [reply app/features-json/projects_json_controller.rb]: Ahh, looks like I'm wrong, it's already renamed

**Comment by taquitos** [comment]
This is neat!
🌮:goat:

**Pull request closed without merging**